services/elasticache/pkg/resource/snapshot/custom_create_api.go

```diff
@@ -0,0 +1,197 @@
+				}
+				if f11iter.NodeGroupConfiguration != nil {
+					f11elemf4 := &svcapitypes.NodeGroupConfiguration{}
+					if f11iter.NodeGroupConfiguration.NodeGroupId != nil {
+						f11elemf4.NodeGroupID = f11iter.NodeGroupConfiguration.NodeGroupId
+					}
+					if f11iter.NodeGroupConfiguration.PrimaryAvailabilityZone != nil {
+						f11elemf4.PrimaryAvailabilityZone = f11iter.NodeGroupConfiguration.PrimaryAvailabilityZone
+					}
+					if f11iter.NodeGroupConfiguration.ReplicaAvailabilityZones != nil {
+						f11elemf4f2 := []*string{}
+						for _, f11elemf4f2iter := range f11iter.NodeGroupConfiguration.ReplicaAvailabilityZones {
+							var f11elemf4f2elem string
+							f11elemf4f2elem = *f11elemf4f2iter
+							f11elemf4f2 = append(f11elemf4f2, &f11elemf4f2elem)
+						}
+						f11elemf4.ReplicaAvailabilityZones = f11elemf4f2
+					}
+					if f11iter.NodeGroupConfiguration.ReplicaCount != nil {
+						f11elemf4.ReplicaCount = f11iter.NodeGroupConfiguration.ReplicaCount
+					}
+					if f11iter.NodeGroupConfiguration.Slots != nil {
+						f11elemf4.Slots = f11iter.NodeGroupConfiguration.Slots
+					}
+					f11elem.NodeGroupConfiguration = f11elemf4
+				}
+				if f11iter.NodeGroupId != nil {
+					f11elem.NodeGroupID = f11iter.NodeGroupId
+				}
+				if f11iter.SnapshotCreateTime != nil {
+					f11elem.SnapshotCreateTime = &metav1.Time{*f11iter.SnapshotCreateTime}
+				}
+				f11 = append(f11, f11elem)
+			}
+			ko.Status.NodeSnapshots = f11
+		}
+		if resp.Snapshot.NumCacheNodes != nil {
+			ko.Status.NumCacheNodes = resp.Snapshot.NumCacheNodes
+		}
+		if resp.Snapshot.NumNodeGroups != nil {
+			ko.Status.NumNodeGroups = resp.Snapshot.NumNodeGroups
+		}
+		if resp.Snapshot.Port != nil {
+			ko.Status.Port = resp.Snapshot.Port
+		}
+		if resp.Snapshot.PreferredAvailabilityZone != nil {
+			ko.Status.PreferredAvailabilityZone = resp.Snapshot.PreferredAvailabilityZone
+		}
+		if resp.Snapshot.PreferredMaintenanceWindow != nil {
+			ko.Status.PreferredMaintenanceWindow = resp.Snapshot.PreferredMaintenanceWindow
+		}
+		if resp.Snapshot.ReplicationGroupDescription != nil {
+			ko.Status.ReplicationGroupDescription = resp.Snapshot.ReplicationGroupDescription
+		}
+		if resp.Snapshot.SnapshotRetentionLimit != nil {
+			ko.Status.SnapshotRetentionLimit = resp.Snapshot.SnapshotRetentionLimit
+		}
+		if resp.Snapshot.SnapshotSource != nil {
+			ko.Status.SnapshotSource = resp.Snapshot.SnapshotSource
+		}
+		if resp.Snapshot.SnapshotStatus != nil {
+			ko.Status.SnapshotStatus = resp.Snapshot.SnapshotStatus
+		}
+		if resp.Snapshot.SnapshotWindow != nil {
+			ko.Status.SnapshotWindow = resp.Snapshot.SnapshotWindow
+		}
+		if resp.Snapshot.TopicArn != nil {
+			ko.Status.TopicARN = resp.Snapshot.TopicArn
+		}
+		if resp.Snapshot.VpcId != nil {
+			ko.Status.VPCID = resp.Snapshot.VpcId
+		}
+
+		rm.setStatusDefaults(ko)
+		// custom set output from response
+		rm.CustomCopySnapshotSetOutput(r, resp, ko)
+		return &resource{ko}, nil
+	}
```

Comment:
where is the implementation when SourceSnapshotName is nil?

Reply:
We would use the auto generated implementation if SourceSnapshotName is nil.

Reply:
Yes, you are correct... I forgot there are two ways of overriding custom stuff. There is the complete override of the custom update method name and then there is the technique of overriding the "implementation" that occurs before the default code paths are taken. We definitely need to clean this up. It's way too confusing. :(